pkg/events/registry.go

```diff
@@ -0,0 +1,37 @@
+package events
+
+import (
+	"k8s.io/klog"
+)
+
+type EventRegistry struct {
+	eventHandlers map[string]EventHandler
+	name          string
+}
+
+func NewRegistry(name string) EventRegistry {
+	return EventRegistry{
+		name: name,
+	}
+}
+
+// Adds a supported driver, prints a fatal error in the case of double registration
+func (er *EventRegistry) AddDriver(name string, eventHandler EventHandler) {
```

Comment:
```suggestion
// AddHandler adds an EventHandler. If one already exists with the handler's name, a fatal error is logged.
func (er *EventRegistry) AddHandler(eventHandler EventHandler) {
```

I assume `name` and `eventHandler.GetName()` should be the same.